From 6f9413ba21c171b4c08795541649adfc16ba75e4 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 10 Jul 2014 01:06:51 -0400
Subject: [acpi] hotplug: Don't set kobject parent pointer explicitly

Message-id: <20140710010651.22146.80473.stgit@gir.stowe>
Patchwork-id: 85622
O-Subject: [RHEL7.1 PATCH 083/105] ACPI / hotplug: Don't set kobject parent pointer explicitly
Bugzilla: 1114228
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1114228
Upstream Status: cae712724161c27f5e8c39bb3946449e1c179881

commit cae712724161c27f5e8c39bb3946449e1c179881
Author: Bjorn Helgaas <bhelgaas@google.com>
Date:   Wed Sep 25 14:23:12 2013 -0600

    ACPI / hotplug: Don't set kobject parent pointer explicitly

    kobject_add() sets the parent pointer, so we don't need to do it
    explicitly.

    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/sysfs.c b/drivers/acpi/sysfs.c
index a3dca13..c8d3abf 100644
--- a/drivers/acpi/sysfs.c
+++ b/drivers/acpi/sysfs.c
@@ -765,7 +765,6 @@ void acpi_sysfs_add_hotplug_profile(struct acpi_hotplug_profile *hotplug,
 
  kobject_init(&hotplug->kobj, &acpi_hotplug_profile_ktype);
 
- hotplug->kobj.parent = hotplug_kobj;
  error = kobject_add(&hotplug->kobj, hotplug_kobj, "%s", name);
  if (error)
   goto err_out;
-- 
1.7.1